official/utils/misc/distribution_utils.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Helper functions for running models in a distributed setting."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+
+
+def get_distribution_strategy(num_gpus, all_reduce_alg=None):
+  """Return a DistributionStrategy for running the model.
+
+  Args:
+    num_gpus: Number of GPUs to run this model.
+    all_reduce_alg: Specify which algorithm to use when performing all-reduce.
```

Comment:
What are the choices and default here? Might be nice to mention going to look at DistributionStrategies if more detail is desired.

Reply:
The choices are `nccl`, `hierarchical_copy`, or if not specified DistStrat will look at the device topology and choose. (hc if it looks like a DGX, otherwise nccl I believe.) I think we want None to be the default.

Reply:
A concern I have with limiting the choices here is that more algorithms may be implemented in the future (and might be a hassle to update our code each time). I'll put a mention of DistStrat here.